batch.R

```diff
@@ -108,10 +98,18 @@ for(i in 1:nrow(fileDF)) {
   qColName <- names(siteConstit)[2]
   dateColName <- names(siteConstit)[1]
   
+  # remove duplicate observations
+  constitDupes <- table(siteConstit$date) %>% .[.>1] %>% names()
+  nonFirstDupes <- unlist(lapply(constitDupes, function(cD) {
+    dupes <- which(as.character(siteConstit$date) == cD)
+    dupes[-1]
```

Comment:
I'm using the feature of R functions that returns the last value created in a function, so `dupes[-1]` from each iteration is getting combined into `nonFirstDupes`. But I'm aware that you and Lindsay and others strive to use `return()` to make these things explicit...it's gonna be a tough habit for me to break, but I'll make that change here (later today) and try to remember in the future.